apcupsd-influxdb-exporter.py

```diff
@@ -5,47 +5,61 @@
 from apcaccess import status as apc
 from influxdb import InfluxDBClient
 
-
 dbname = os.getenv('INFLUXDB_DATABASE', 'apcupsd')
 user = os.getenv('INFLUXDB_USER')
 password = os.getenv('INFLUXDB_PASSWORD')
 port = os.getenv('INFLUXDB_PORT', 8086)
-host = os.getenv('INFLUXDB_HOST', 'localhost')
-interval = float(os.getenv('INTERVAL', 5))
-client = InfluxDBClient(host, port, user, password, dbname)
+host = os.getenv('INFLUXDB_HOST')
+apcupsdHost = os.getenv('APCUPSD_HOST', host)
+
+delay = os.getenv('DELAY', 10)
+
+removeTheseKeys = ['DATE', 'STARTTIME', 'END APC']
+tagKeys = ['APC', 'HOSTNAME', 'UPSNAME', 'VERSION', 'CABLE', 'MODEL', 'UPSMODE', 'DRIVER', 'APCMODEL', 'END APC']
 
+wattsKey = 'WATTS'
+nominalPowerKey = 'NOMPOWER'
+
+client = InfluxDBClient(host, port, user, password, dbname)
 client.create_database(dbname)
 
 while True:
-    ups = apc.parse(apc.get(host=os.getenv('APCUPSD_HOST', 'localhost')), strip_units=True)
-
-    watts = float(os.getenv('WATTS', ups.get('NOMPOWER', 0.0))) * 0.01 * float(ups.get('LOADPCT', 0.0))
-    json_body = [
-        {
-            'measurement': 'apcaccess_status',
-            'fields': {
-                'WATTS': watts,
-                'STATUS': ups.get('STATUS'),
-                'LOADPCT': float(ups.get('LOADPCT', 0.0)),
-                'BCHARGE': float(ups.get('BCHARGE', 0.0)),
-                'TONBATT': float(ups.get('TONBATT', 0.0)),
-                'TIMELEFT': float(ups.get('TIMELEFT', 0.0)),
-                'NOMPOWER': float(ups.get('NOMPOWER', 0.0)),
-                'CUMONBATT': float(ups.get('CUMONBATT', 0.0)),
-                'BATTV': float(ups.get('BATTV', 0.0)),
-                'OUTPUTV': float(ups.get('OUTPUTV', 0.0)),
-                'ITEMP': float(ups.get('ITEMP', 0.0)),
-            },
-            'tags': {
-                'host': os.getenv('HOSTNAME', ups.get('HOSTNAME', 'apcupsd-influxdb-exporter')),
-                'serial': ups.get('SERIALNO', None),
+    try:
+        ups = apc.parse(apc.get(host=apcupsdHost), strip_units=True)
+        for key in removeTheseKeys:
+            del ups[key]
+
+        tags = {'host': os.getenv('HOSTNAME', ups.get('HOSTNAME', 'apcupsd-influxdb-exporter'))}
+
+        for key in tagKeys:
+            if key in ups:
+                tags[key] = ups[key]
+                del ups[key]
+
+        if wattsKey not in os.environ and nominalPowerKey not in ups:
+            raise ValueError("Your UPS does not specify NOMPOWER, you must specify the max watts your UPS can produce.")
+
+        for key in ups:
+            if ups[key].replace('.', '', 1).isdigit():
+                ups[key] = float(ups[key])
+
+        ups[wattsKey] = float(os.getenv('WATTS', ups.get('NOMPOWER', 0.0))) * 0.01 * float(ups.get('LOADPCT', 0.0))
```

Comment:
ups[wattsKey] = float(os.getenv(wattsKey, ups.get(nominalPowerKey))) * 0.01 * float(ups.get('LOADPCT', 0.0))

Using the variables you created above for the key names. I think you can also get rid of the default return on the ups.get(nominalPowerKey) key call, because either you'll have os.getenv(wattsKey) return something, or you know ups.get(nominalPowerKey) will return something. Your guard clause up above that raises a ValueError guarantees that